### PR TITLE
Combined the datastore interface of the remote and the cache.

### PR DESCRIPTION
**Pull request by pokk**
as the title

### CODE REVIEW

**Comment by pokk** [comment]
@lekaha 

Please take a look.

BTW, please ignore the code in the cache layer, they're temporary.

File: data/src/main/java/com/playone/mobile/data/repository/PlayoneCache.kt [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
-import com.playone.mobile.ext.invalidInt
 import io.reactivex.Completable
-import io.reactivex.Single
 
 /**
  * Interface defining methods for the caching of [PlayoneItem]. This is to be implemented by the
  * cache layer, using this interface as a way of communicating.
  */
-interface PlayoneCache : CacheChecker {
+interface PlayoneCache : PlayoneRemote {
```
Comment (lekaha) [inline data/src/main/java/com/playone/mobile/data/repository/PlayoneCache.kt:12]: I think you can extend both `CachChecker` and  `PlayoneRemote`. But I feel `PlayoneRemote` should not be super-interface of `PlayoneCache` (I see super-interface is the same as parent class)

Comment (lekaha) [reply data/src/main/java/com/playone/mobile/data/repository/PlayoneCache.kt:12]: Btw, I feel ok if they are independent. I know you want to make them having the same interface but you can think about extract off the common interfaces from both and be extended by `PlayoneCache` and `PlayoneRemote`. But just want you know `CacheChecker` seems is needed.

Comment (pokk) [reply data/src/main/java/com/playone/mobile/data/repository/PlayoneCache.kt:12]: You're right!
Here I'll fix the interface again, please allow me to put it here temporarily. Becz using `CacheChcker` isn't a good way to implement a cache checking mechanism.

**Pull request merged**